File: symfony/src/Entity/Cards/DeckTagAssignment.php
<?php

namespace App\Entity\Cards;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use App\Repository\Cards\DeckTagAssignmentRepository;

#[ORM\Entity(repositoryClass: DeckTagAssignmentRepository::class)]
#[ORM\Table(name: 'deck_tag_assignment')]
class DeckTagAssignment
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups(['deckTagAssignment:read'])]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Deck::class, inversedBy: 'tag_assignments')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Deck $deck = null;

    #[ORM\ManyToOne(targetEntity: DeckTag::class, inversedBy: 'deck_assignments')]
    #[ORM\JoinColumn(nullable: false)]
    private ?DeckTag $tag = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    #[Groups(['deckTagAssignment:read'])]
    public function getDeckId(): ?int
    {
        return $this->deck?->getId();
    }

    public function getDeck(): ?Deck
    {
        return $this->deck;
    }

    public function setDeck(?Deck $deck): static
    {
        $this->deck = $deck;
        return $this;
    }

    #[Groups(['deckTagAssignment:read'])]
    public function getTagId(): ?int
    {
        return $this->tag?->getId();
    }

    public function getTag(): ?DeckTag
    {
        return $this->tag;
    }

    public function setTag(?DeckTag $tag): static
    {
        $this->tag = $tag;
        return $this;
    }

}

File: spring/src/main/java/cardsproject/domain/cards/DeckTagAssignment.java
package cardsproject.domain.cards;

import jakarta.persistence.*;

@Entity
@Table(name = "deck_tag_assignments")
public class DeckTagAssignment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    // @ManyToOne -> Deck, onDelete=CASCADE, relatedName=tag_assignments
    @Column(name = "deck_id")
    private Long deckId;
    // @ManyToOne -> DeckTag, onDelete=CASCADE, relatedName=deck_assignments
    @Column(name = "tag_id")
    private Long tagId;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getDeckId() { return deckId; }
    public void setDeckId(Long deckId) { this.deckId = deckId; }
    public Long getTagId() { return tagId; }
    public void setTagId(Long tagId) { this.tagId = tagId; }
}
